File: CS/XPO/NET_Framework/RuntimeDbChooser.Web/Controllers/CustomLogonController.cs
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.Web;
using RuntimeDbChooser.Module.BusinessObjects;

namespace RuntimeDbChooser.Web.Controllers {
    public class CustomLogonController : WebLogonController {
        protected override void Accept(SimpleActionExecuteEventArgs args) {
            string targetDataBaseName = ((IDatabaseNameParameter)Application.Security.LogonParameters).DatabaseName;
            string newConnectionString = MSSqlServerChangeDatabaseHelper.PatchConnectionString(targetDataBaseName, Application.ConnectionString);
            if(Application.ConnectionString != newConnectionString) {
                Application.ObjectSpaceProviderContainer?.Clear();
                Application.ConnectionString = newConnectionString;
            }
            base.Accept(args);
        }
    }
}

File: VB/XPO/NET_Framework/RuntimeDbChooser.Web/Controllers/CustomLogonController.vb
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.Web
Imports RuntimeDbChooser.Module.BusinessObjects

Namespace RuntimeDbChooser.Web.Controllers

    Public Class CustomLogonController
        Inherits WebLogonController

        Protected Overrides Sub Accept(ByVal args As SimpleActionExecuteEventArgs)
            Dim targetDataBaseName As String = CType(Application.Security.LogonParameters, IDatabaseNameParameter).DatabaseName
            Dim newConnectionString As String = MSSqlServerChangeDatabaseHelper.PatchConnectionString(targetDataBaseName, Application.ConnectionString)
            If Not Equals(Application.ConnectionString, newConnectionString) Then
                Application.ObjectSpaceProviderContainer?.Clear()
                Application.ConnectionString = newConnectionString
            End If

            MyBase.Accept(args)
        End Sub
    End Class
End Namespace
